test.js

```diff
@@ -7,668 +7,465 @@ const Next = require('next')
+  }
 
-  await Promise.all(commonAssets.map(suffix => testNextAsset(t, fastify, `/base_path/_next/${suffix}`)))
+  fastify.close()
 })
 
 test('should not serve static assets with provided option noServeAssets: true', async t => {
-  const manifest = require('./.next/build-manifest.json')
-
-  const fastify = Fastify()
+  const fastify = await Fastify().register(require('./index'), {
+    noServeAssets: true,
+    underPressure: false
+  })
+  fastify.next('/hello')
 
-  fastify
-    .register(require('./index'), {
-      noServeAssets: true,
-      underPressure: false
-    })
-    .after(() => {
-      fastify.next('/hello')
-    })
+  const origin = await fastify.listen({ port })
+  const manifest = require('./.next/build-manifest.json')
+  const assets = manifest.pages['/hello']
 
-  t.teardown(() => fastify.close())
+  t.ok(assets.length > 0)
 
-  const commonAssets = manifest.pages['/hello']
+  for await (const asset of assets) {
+    const { statusCode, headers } = await request({ path: `/_next/${asset}`, origin })
 
-  t.ok(commonAssets.length > 0)
+    t.equal(statusCode, 404)
+    t.equal(headers['content-type'], 'application/json; charset=utf-8')
+  }
 
-  await Promise.all(commonAssets.map(suffix => testNoServeNextAsset(t, fastify, `/_next/${suffix}`)))
+  fastify.close()
 })
 
-test('should return a json data on api route', t => {
-  t.plan(3)
+test('should return a json data on api route', async t => {
+  const fastify = await Fastify().register(require('./index'))
+  fastify.next('/api/*')
 
-  const fastify = Fastify()
-  fastify
-    .register(require('./index'))
-    .after(() => {
-      fastify.next('/api/*')
-    })
+  const origin = await fastify.listen({ port })
+  const { statusCode, headers } = await request({ path: '/api/user', origin })
 
-  t.teardown(() => fastify.close())
+  t.equal(statusCode, 200)
+  t.equal(headers['content-type'], 'application/json')
 
-  fastify.inject({
-    url: '/api/user',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'application/json')
-  })
+  fastify.close()
 })
 
-test('should not log any errors', t => {
-  t.plan(5)
+test('should not log any errors', async t => {
+  let didLog = false
 
-  let showedError = false
   const logger = pino({
     level: 'error',
-    formatters: {
-      log: (obj) => {
-        showedError = true
-        return obj
-      }
-    }
-  })
-
-  const fastify = Fastify({
-    logger
-  })
-
-  fastify
-    .register(require('./index')).after(() => {
-      fastify.next('/hello')
-    })
-
-  fastify.inject({
-    url: '/hello',
-    method: 'GET'
-  }, (err, res) => {
-    t.error(err)
-    t.equal(res.statusCode, 200)
-    t.equal(res.headers['content-type'], 'text/html; charset=utf-8')
-    t.match(res.payload, '<div>hello world</div>')
-    t.equal(showedError, false, 'Should not show any error')
-  })
-})
-
-test('should respect plugin logLevel', t => {
```

Comment:
this test is gone. why?

Reply:
It didn't actually provide any extra coverage or specific assertion after the 'should not log any errors' test.